ompi/mca/coll/base/coll_base_alltoallv.c

```diff
@@ -281,8 +281,10 @@ ompi_coll_base_alltoallv_intra_basic_linear(const void *sbuf, const int *scounts
         for( i = 0; i < nreqs; i++ ) {
             if (MPI_REQUEST_NULL == reqs[i]) continue;
             if (MPI_ERR_PENDING == reqs[i]->req_status.MPI_ERROR) continue;
-            err = reqs[i]->req_status.MPI_ERROR;
-            break;
+            if (reqs[i]->req_status.MPI_ERROR) {
```

Comment:
Please be explicit:

```suggestion
            if (reqs[i]->req_status.MPI_ERROR != MPI_SUCCESS) {
```

Reply:
Is it supposed to be (MPI_SUCCESS != reqs[i]->req_status.MPI_ERROR) to comply with ompi convention, or it does not matter?

Reply:
Good catch.

It can go either way -- it's `==` that's more the intent of the "constants on the left" style (vs. `!=`), because `foo = MPI_SUCCESS` and `foo == MPI_SUCCESS` are both legal and have very different meanings.  Putting the constant on the left will catch the error, because `MPI_SUCCESS = foo` will generate a compile errors.

`!=` does not share this property -- `MPI_SUCCESS != foo` and `foo != MPI_SUCCESS` will both compile.  And it's harder to get wrong than `=` vs. `==`.